api/src/main/java/net/kyori/adventure/bossbar/BossBar.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.bossbar;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.util.NameMap;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * A bossbar.
+ */
+public interface BossBar {
+  /**
+   * Gets the name.
+   *
+   * @return the name
+   */
+  @NonNull Component name();
+
+  /**
+   * Sets the name.
+   *
+   * @param name the name
+   * @return the bossbar
+   */
+  @NonNull BossBar name(final @NonNull Component name);
+
+  /**
+   * Gets the percent.
+   *
+   * @return the percent
+   */
+  float percent();
+
+  /**
+   * Sets the percent.
+   *
+   * @param percent the percent
```

Comment:
Should the api call out this is from [0, 1]. Notchian clients render > 1 weirdly.